### PR TITLE
Document laravel provider

### PR DESCRIPTION
**Pull request by soyuka**

### CODE REVIEW

File: laravel/index.md [hunk 1]
```diff
@@ -165,9 +165,65 @@ So, if you want to access the raw data, you have two alternatives:
 
 For instance, go to `http://127.0.0.1:8000/api/books.jsonld` to retrieve the list of `Book` resources in JSON-LD.
 
+> [!NOTE] Read the next parameter if you want to use JSON:API instead!
```
Comment (dunglas) [inline laravel/index.md:168]:
```suggestion
> [!NOTE]
> Documentation for Eloquent "API resources" encourages using the JSON:API community format.
> While we recommend preferring JSON-LD when possible, JSON:API is also supported by API Platform,
> read the [Content Negotiation](#content-negotiation) section to learn how to enable it.
```

File: laravel/index.md [hunk 2]
```diff
@@ -165,9 +165,65 @@ So, if you want to access the raw data, you have two alternatives:
...
+
 Of course, you can also use your favorite HTTP client to query the API.
 We are fond of [Hoppscotch](https://hoppscotch.com), a free and open source API client with good support of API Platform.
 
+As recommended by our [design considerations](../core/design.md), you can totally use the data source of your choice using a [provider](../core/state-providers.md):
```
Comment (dunglas) [inline laravel/index.md:173]:
```suggestion

## Using Data Transfer Objects and Hooking Custom Logic

While exposing directly the data in the database is convenient for Rapid Application Development, using different classes for the internal data and the public data is a good practice for more complex projects. 

As explained in our [general design considerations](../core/design.md), API Platform allows us to use the data source of our choice using a [provider](../core/state-providers.md) and Data Transfer Objects (DTOs) are first-class citizens:
```

File: laravel/index.md [hunk 3]
```diff
@@ -165,9 +165,65 @@ So, if you want to access the raw data, you have two alternatives:
...
+
+```php
+<?php
+
+namespace App\State;
+
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProviderInterface;
+
+final class BookProvider implements ProviderInterface
```
Comment (dunglas) [inline laravel/index.md:183]: Maybe also show an example of how to inject Eloquent in the data provider to hydrate a DTO?

File: laravel/index.md [hunk 4]
```diff
@@ -165,9 +165,65 @@ So, if you want to access the raw data, you have two alternatives:
...
 For instance, go to `http://127.0.0.1:8000/api/books.jsonld` to retrieve the list of `Book` resources in JSON-LD.
 
+> [!NOTE] Read the next parameter if you want to use JSON:API instead!
+
 Of course, you can also use your favorite HTTP client to query the API.
 We are fond of [Hoppscotch](https://hoppscotch.com), a free and open source API client with good support of API Platform.
 
+As recommended by our [design considerations](../core/design.md), you can totally use the data source of your choice using a [provider](../core/state-providers.md):
+
+```php
```
Comment (dunglas) [inline laravel/index.md:175]: Maybe could you create a DTO in an `App\ApiResource` namespace and update the config to explain how to register this directory?

**Review by dunglas — CHANGES_REQUESTED** [review]

**Review by dunglas — APPROVED** [review]

File: laravel/index.md [hunk 5]
```diff
@@ -165,9 +165,116 @@ So, if you want to access the raw data, you have two alternatives:
...
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProviderInterface;
+use App\Models\Book as BookModel;
+
+final class BookProvider implements ProviderInterface
+{
+    public function provide(Operation $operation, array $uriVariables = [], array $context = []): object|array|null
+    {
+        $book = BookModel::find($uriVariables['id']);
+        return new Book(id: $book->id, title: $book->title);
```
Comment (EMeDesign) [inline laravel/index.md:232]: Hi! Do you forget to import Book Dto?

`use App/ApiResource/Book`

Comment (vinceAmstoutz) [reply laravel/index.md:232]: This should be solved in #2036, thanks!

**Pull request merged**